src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2344,6 +2959,25 @@ def __init__(self, cfg, logger):
                 raise ValueError("Not enough cores available (%s) to satisfy allocation request (%s)." \
                                 % (str(cores_avail), str(self.requested_cores)))
 
+        # NOTE: self.lrms_info is what scheduler and launch method can
+        # ultimately use, as it is included into the cfg passed to all
+        # components.
+        #
+        # four elements are well defined:
+        #   lm_info:        the dict received via the LM's lrms_config_hook
+        #   node_list:      a list of node names to be used for unit execution
+        #   cores_per_node: as the name says
+        #   agent_nodes:    list of node names reserved for agent execution
+        #
+        # That list may turn out to be insufficient for some schedulers.  Yarn
+        # for example may need to communicate YARN service endpoints etc.  an
+        # LRMS can thus expand this dict, but is then likely bound to a specific
+        # scheduler which can interpret the additional information.
+        self.lrms_info['lm_info']        = self.lm_info
+        self.lrms_info['node_list']      = self.node_list
+        self.lrms_info['cores_per_node'] = self.cores_per_node
+        self.lrms_info['agent_nodes']    = self.agent_nodes
+
```

Comment:
I am not sure I understand this  diff.  (a): all those details are already on the `self._cfg` I believe, why is it stored again?   (b) where is self.lrms_info used?  I can't find it referenced...

I think this setting is not needed, since (as you say yourself) the information is passed via the cfg?